frontend/src/services/web3-provider/helper.ts

```diff
@@ -0,0 +1,31 @@
+import { useRpcUrls } from '@/composables/useRpcUrls';
+import { useConfiguration } from '@/stores/configuration';
+import { useSettings } from '@/stores/settings';
+import { WalletType } from '@/types/settings';
+
+import { createMetaMaskProvider, createWalletConnectProvider } from './';
+import { EthereumProvider } from './types';
+
+export async function getConnectedWalletProvider(): Promise<EthereumProvider | undefined> {
+  const configuration = useConfiguration();
+  const settings = useSettings();
+  const rpcUrlsList = useRpcUrls(configuration.chains);
```

Comment:
same as the other comment: stores and composables should not be accessed in helper functions.